src/blenderbim/blenderbim/bim/module/pset/operator.py

```diff
@@ -374,3 +374,147 @@ def _execute(self, context):
                 prop_name=self.name,
                 prop_value=prop_value,
             )
+
+
+class AddPropertyToEdit(bpy.types.Operator):
+    bl_label = "Add new row"
+    bl_idname = "bim.add_property_to_edit"
+    bl_options = {"REGISTER", "UNDO"}
+    type: bpy.props.StringProperty()
+
+    def execute(self, context):
+        if self.type == "PropertiesToRename":
+            props = context.scene.PropertiesToRename
```

Comment:
Since `type` string value is the same as the scene's custom prop name, maybe doing `getattr(context.scene, self.type).add()` is simpler ? It also makes operators easier to extend if somehow you want to add other modes later. Matter of preference :)

Reply:
Agreed